src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -30,34 +30,52 @@ namespace jit {
 namespace dsl {
 
 struct ctx_t {
+    bool new_ir_api() const { return new_ir_api_; }
 
-    void declare_kernel(const kernel_iface_t &interface, ir_context_t &ctx) {
+    void declare_kernel(const kernel_iface_t &interface, ir_context_t &ctx,
+            bool new_ir_api = false) {
+        slm_byte_offset_ = 0;
+        new_ir_api_ = new_ir_api;
         gpu_assert(stmts_stack_.empty())
                 << "Invalid generation of a kernel within a kernel";
         interface_ = interface;
         ctx_ = &ctx;
 
         begin_scope();
 
-        for (int i = 0; i < interface.nargs(); i++) {
-            const auto &var = interface.arg_var(i);
-            if (var.type().is_ptr()) {
-                if (var.type().is_slm()) {
-                    append(alloc_t::make(var, 0, alloc_kind_t::slm, stmt_t {}));
+        if (new_ir_api_) {
+            for (int i = 0; i < 3; i++) {
+                local_sizes_[i] = var_t::make(local_size_type(),
+                        std::string("local_size_") + "012"[i]);
+                local_ids_[i] = var_t::make(
+                        local_id_type(), std::string("local_id_") + "012"[i]);
+                group_ids_[i] = var_t::make(
+                        group_id_type(), std::string("group_id_") + "012"[i]);
+            }
```

Comment:
How are we going to handle logging kernel interfaces under this new API?

Reply:
Good question, we need to think about it. I would add a section with an ordered list of arguments in the beginning of a kernel output.